src/api/comms.test.js

```diff
@@ -32,6 +43,80 @@ describe('getHeaders', () => {
   });
 });
 
+describe('getPatchHeaders', () => {
+  it('returns default headers when called with no params', () => {
+    expect(getPatchHeaders()).not.toBeNull();
+  });
+
+  it('combines custom headers with the default', () => {
+    const customHeaders = {
+      'X-Foo': 'Bar'
+    };
+    const result = getPatchHeaders(customHeaders);
+    expect(result).toMatchObject(customHeaders);
+    expect(result).toMatchObject(getPatchHeaders());
+  });
+});
+
+describe('patchAddSecretBody', () => {
+  it('patchAddSecretBody', () => {
+    const secretName = 'Groot';
+    const secretResponse = [
+      {
+        op: 'add',
+        path: 'serviceaccount/secrets/-',
+        value: {
+          name: secretName
+        }
+      }
+    ];
+    const result = patchAddSecretBody(secretName);
+    expect(result).toMatchObject(secretResponse);
+    expect(result).toMatchObject(patchAddSecretBody(secretName));
+  });
+});
+
+describe('patchAddSecret', () => {
+  it('patchAddSecret', () => {
+    const data = {
+      fake: 'data'
+    };
+    fetchMock.mock(uri, data);
+    return patchAddSecret(uri, data).then(response => {
+      expect(response).toEqual(data);
+      fetchMock.restore();
+    });
+  });
+});
+
+describe('patchRemoveSecretBody', () => {
+  it('patchRemoveSecretBody', () => {
```

Comment:
Not the easiest of tests to read and comprehend, can you add a comment here/appropriate test name so it's more clear/easier to maintain?

Reply:
Done 😸